src/wp-admin/includes/class-theme-upgrader.php

```diff
@@ -566,15 +566,27 @@ public function check_package( $source ) {
 			);
 		}
 
-		// If it's not a child theme, it must have at least an index.php to be legit.
-		if ( empty( $info['Template'] ) && ! file_exists( $working_directory . 'index.php' ) ) {
+		/*
+		 * Parent themes must contain an index file:
+		 * - classic themes require /index.php
+		 * - block themes require /templates/index.html
+		 */
+		if (
+			empty( $info['Template'] ) &&
+			! file_exists( $working_directory . 'index.php' ) &&
+			! file_exists( $working_directory . 'templates/index.html' )
```

Comment:
```suggestion
			! file_exists( $working_directory . 'templates/index.html' ) &&
			! file_exists( $working_directory . 'block-templates/index.html' )
```

The PR should also account for `block-templates/index.html`, which is accounted for in [WP_Theme::is_block_theme()](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/class-wp-theme.php#L1484) for BC reasons [(see explanation)](https://github.com/WordPress/wordpress-develop/blob/a11fd0614d36a79321d1480cad547c763ae433f3/src/wp-includes/block-template-utils.php#L23).